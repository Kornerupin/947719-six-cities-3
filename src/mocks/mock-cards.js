export default [
  {
    type: `Apartment`,
    price: `80`,
    name: `Beautiful &amp; luxurious apartment at great location`
  },
  {
    type: `Private room`,
    price: `132`,
    name: `Wood and stone place`
  },
  {
    type: `Apartment`,
    price: `180`,
    name: `Canal View Prinsengracht`
  },
  {
    type: `Apartment`,
    price: `80`,
    name: `Nice, cozy, warm big bed apartment`
  },
];
