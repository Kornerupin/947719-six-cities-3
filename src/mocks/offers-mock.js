export default [
  {
    id: 1,
    city: `Amsterdam`,
    image: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    type: `Apartment`,
    price: `80`,
    rating: 3.1,
    name: `Beautiful &amp; luxurious apartment at great location`,
    coords: [52.3909553943508, 4.85309666406198],
    reviews: [{
      avatar: `https://api.adorable.io/avatars/128/0.9912918678423135`,
      name: `Max`,
      rating: 4.1,
      text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      time: 1569006474313
    },
    {
      avatar: `https://api.adorable.io/avatars/128/0.33908914729720996`,
      name: `Sax`,
      rating: 4.5,
      text: `It is my favorite hotel. I have been staying her for several years and can confirm that service level remains at the same high level.`,
      time: 1571598485610
    },
    {
      avatar: `https://api.adorable.io/avatars/128/0.06997580364137201`,
      name: `Rax`,
      rating: 3.8,
      text: `Emplacement parfait, beaucoup de commerces à proximité, le tout dans un lieu traditionnel et agréable, j'ai adoré mon court séjour et aimerait beaucoup y retourner un jour!`,
      time: 1574276894250
    }]
  },
  {
    id: 2,
    city: `Amsterdam`,
    image: [`img/room.jpg`],
    type: `Private room`,
    price: `132`,
    rating: 1.8,
    name: `Wood and stone place`,
    coords: [52.369553943508, 4.85309666406198],
    reviews: [{
      avatar: `https://api.adorable.io/avatars/128/0.3963321309597061`,
      name: `Wax`,
      rating: 3.8,
      text: `Very nice hotel, helpful and friendly staff (thank you guys for everything), good breakfast, clean and comfy room.`,
      time: 1576868903507
    }]
  },
  {
    id: 3,
    city: `Amsterdam`,
    image: [`img/apartment-02.jpg`],
    type: `Apartment`,
    price: `180`,
    rating: 3.8,
    name: `Canal View Prinsengracht`,
    coords: [52.3909553943508, 4.929309666406198],
    reviews: [{
      avatar: `https://api.adorable.io/avatars/128/0.1138720045992132`,
      name: `Tax`,
      rating: 4.5,
      text: `Breakfast was excellent. Everything was really tasty. Location perfect, in a walking distance from everything. Staff was friendly and welcoming. Hotel was really silent.`,
      time: 1576868903507
    }]
  },
  {
    id: 4,
    city: `Amsterdam`,
    image: [`img/apartment-03.jpg`],
    type: `Apartment`,
    price: `80`,
    rating: 4.5,
    name: `Nice, cozy, warm big bed apartment`,
    coords: [52.3809553943508, 4.939309666406198],
    reviews: [{
      avatar: `https://api.adorable.io/avatars/128/0.477232804299224`,
      name: `Bax`,
      rating: 3.2,
      text: `The only drawback for me was that I could hear the traffic through the window in my room. It was therefore not often i opened the window to get fresh air.`,
      time: 1576868903507
    }]
  },
];
