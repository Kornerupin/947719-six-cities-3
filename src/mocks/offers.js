export default [
  {
    image: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    type: `Apartment`,
    price: `80`,
    rating: 3.1,
    name: `Beautiful &amp; luxurious apartment at great location`,
    coords: [52.3909553943508, 4.85309666406198]
  },
  {
    image: [`img/room.jpg`],
    type: `Private room`,
    price: `132`,
    rating: 1.8,
    name: `Wood and stone place`,
    coords: [52.369553943508, 4.85309666406198]
  },
  {
    image: [`img/apartment-02.jpg`],
    type: `Apartment`,
    price: `180`,
    rating: 3.8,
    name: `Canal View Prinsengracht`,
    coords: [52.3909553943508, 4.929309666406198]
  },
  {
    image: [`img/apartment-03.jpg`],
    type: `Apartment`,
    price: `80`,
    rating: 4.5,
    name: `Nice, cozy, warm big bed apartment`,
    coords: [52.3809553943508, 4.939309666406198]
  },
];
