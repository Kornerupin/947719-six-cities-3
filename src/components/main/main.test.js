import Main from './main';
import {offersMock} from '../mocks/offers';

it(`<Main /> should render <Offers />`, () => {
  const tree = renderer.create(<Main
    offers={offersMock}
    onOfferMouseOver={() => { }}
    onOfferNameClick={() => { }}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
