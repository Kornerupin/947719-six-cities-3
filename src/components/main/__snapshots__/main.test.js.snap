// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render offers 1`] = `
<div
  classNameName="page page--gray page--main"
>
  <header
    classNameName="header"
  >
    <div
      classNameName="container"
    >
      <div
        classNameName="header__wrapper"
      >
        <div
          classNameName="header__left"
        >
          <a
            classNameName="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              classNameName="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          classNameName="header__nav"
        >
          <ul
            classNameName="header__nav-list"
          >
            <li
              classNameName="header__nav-item user"
            >
              <a
                classNameName="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  classNameName="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  classNameName="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    classNameName="page__main page__main--index true"
  >
    <h1
      classNameName="visually-hidden"
    >
      Cities
    </h1>
    <div
      classNameName="tabs"
    >
      <section
        classNameName="locations container"
      >
        <ul
          classNameName="locations__list tabs__list"
        >
          <li
            classNameName="locations__item"
          >
            <a
              classNameName="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            classNameName="locations__item"
          >
            <a
              classNameName="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            classNameName="locations__item"
          >
            <a
              classNameName="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            classNameName="locations__item"
          >
            <a
              classNameName="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            classNameName="locations__item"
          >
            <a
              classNameName="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            classNameName="locations__item"
          >
            <a
              classNameName="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      classNameName="cities"
    >
      <div
        classNameName="cities__places-container container"
      >
        <section
          classNameName="cities__places places"
        >
          <h2
            classNameName="visually-hidden"
          >
            Places
          </h2>
          <b
            classNameName="places__found"
          >
            2 places to stay in Amsterdam
          </b>
          <div
            classNameName="cities__places-list places__list tabs__content"
          >
            <article
              classNameName="cities__place-card place-card"
              onMouseOver={[Function]}
            >
              <div
                classNameName="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    classNameName="place-card__image"
                    height="200"
                    src="img/apartment-01.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                classNameName="place-card__info"
              >
                <div
                  classNameName="place-card__price-wrapper"
                >
                  <div
                    classNameName="place-card__price"
                  >
                    <b
                      classNameName="place-card__price-value"
                    >
                      €
                      80
                    </b>
                    <span
                      classNameName="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    classNameName="place-card__bookmark-button place-card__bookmark-button--active button"
                    type="button"
                  >
                    <svg
                      classNameName="place-card__bookmark-icon"
                      height="19"
                      id="icon-bookmark"
                      viewBox="0 0 17 18"
                      width="18"
                    >
                      <path
                        d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
                      />
                    </svg>
                    <span
                      classNameName="visually-hidden"
                    >
                      In bookmarks
                    </span>
                  </button>
                </div>
                <div
                  classNameName="place-card__rating rating"
                >
                  <div
                    classNameName="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      classNameName="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  classNameName="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    Beautiful &amp; luxurious apartment at great location
                  </a>
                </h2>
                <p
                  classNameName="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
            <article
              classNameName="cities__place-card place-card"
              onMouseOver={[Function]}
            >
              <div
                classNameName="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    classNameName="place-card__image"
                    height="200"
                    src="img/room.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                classNameName="place-card__info"
              >
                <div
                  classNameName="place-card__price-wrapper"
                >
                  <div
                    classNameName="place-card__price"
                  >
                    <b
                      classNameName="place-card__price-value"
                    >
                      €
                      132
                    </b>
                    <span
                      classNameName="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    classNameName="place-card__bookmark-button place-card__bookmark-button--active button"
                    type="button"
                  >
                    <svg
                      classNameName="place-card__bookmark-icon"
                      height="19"
                      id="icon-bookmark"
                      viewBox="0 0 17 18"
                      width="18"
                    >
                      <path
                        d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
                      />
                    </svg>
                    <span
                      classNameName="visually-hidden"
                    >
                      In bookmarks
                    </span>
                  </button>
                </div>
                <div
                  classNameName="place-card__rating rating"
                >
                  <div
                    classNameName="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "40%",
                        }
                      }
                    />
                    <span
                      classNameName="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  classNameName="place-card__name"
                  onClick={[Function]}
                >
                  <a
                    href="#"
                  >
                    Wood and stone place
                  </a>
                </h2>
                <p
                  classNameName="place-card__type"
                >
                  Private room
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          classNameName="cities__right-section"
        >
          <section
            classNameName="cities__map map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
